### PR TITLE
[#1996] App Store

### LINKED ISSUE #1996: Add App Store to the website
**Task:**
- Add `App store` to the website

**Design:**
- https://www.figma.com/file/zbQ5xuZqAfhAlOiy9ilZlm/App-Store

Let me know if you need any clarification on the task.

---
#### Update
Updated Design: https://www.figma.com/file/zbQ5xuZqAfhAlOiy9ilZlm/App-Store?node-id=771%3A7525

### PR DESCRIPTION
**Pull request by jamessspanggg**
Fixes #1996

- [x] App Store Home 
- [x] App Details Page
- [x] Hook up with API
- [x] Image Slider
- [x] Update based on the final responsive design 
- [x] Styled components

### CODE REVIEW

**Comment by jamessspanggg** [comment]
Note: to merge after @manishram updates the list of apps

File: src/containers/AppStore/components/AppCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  };
+
+  return (
+    <S.Container role="button" onClick={handleClickCard} tabIndex={0}>
+      <S.Banner alt={title} className="AppCard__banner" loading="lazy" src={bannerUrl} />
+      <S.BottomContainer className="AppCard__bottom-container">
+        <S.DetailsContainer className="AppCard__details-container">
+          <S.AppTitleDescriptionContainer className="AppCard__app-title-description-container">
+            <S.AppTitle className="AppCard__app-title">{title}</S.AppTitle>
+            <S.AppDescription className="AppCard__app-description">{description}</S.AppDescription>
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/components/AppCard/index.tsx:34]: what's the use of classNames if we're using styled-components?

File: src/containers/AppStore/components/ImageCarousel/Styles.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
+import styled from 'styled-components';
+
+import colors from 'styles/colors';
+
+import {Carousel as ReactCarousel} from 'react-responsive-carousel';
+import {Icon} from '@thenewboston/ui';
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/components/ImageCarousel/Styles.ts:6]: 3rd party imports should come first

File: src/containers/AppStore/components/ImageCarousel/Styles.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  @media (max-width: 768px) {
+    max-width: 100%;
+    padding: 0 ${sidePadding768};
+  }
+`;
+
+export const Thumbnail = styled.img`
+  border: 4px solid transparent;
+  cursor: pointer;
+  height: 93px;
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/components/ImageCarousel/Styles.ts:84]: would be nice to use multiples of 4 or 8

File: src/containers/AppStore/containers/AppDetails/AppDetailsOverview/index.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import React, {FC} from 'react';
+
+import * as S from './Styles';
+
+type Props = {
+  className?: string;
+  description: string;
+};
+
+const AppDetailsOverview: FC<Props> = ({className, description}) => {
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/containers/AppDetails/AppDetailsOverview/index.tsx:10]: we can use `SFC` here.

File: src/components/Avatar/index.tsx [hunk 5]
```diff
@@ -27,14 +27,18 @@ export const getFormattedSrc = (src: string, size: number): string => {
   }
 };
 
+export type Shape = 'circle' | 'square';
+
 export interface AvatarProps {
   bordered?: boolean;
+  className?: string;
```
Comment (thesanjeevsharma) [inline src/components/Avatar/index.tsx:34]: doesn't `SFC` take care of adding `className` as prop?

File: src/containers/AppStore/containers/AppDetails/AppDetailsSlider/index.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import ImageCarousel from '../../../components/ImageCarousel';
+
+import * as S from './Styles';
+
+type Props = {
+  className?: string;
+  imageUrls: string[];
+};
+
+const AppDetailsSlider: FC<Props> = ({className, imageUrls}) => {
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/containers/AppDetails/AppDetailsSlider/index.tsx:12]: ~`FC`~ `SFC`

File: src/containers/AppStore/containers/AppDetails/AppDetailsTopSection/Styles.ts [hunk 7]
```diff
@@ -0,0 +1,86 @@
+import styled from 'styled-components';
+import {Container as SharedContainer, Avatar as SharedAvatar} from 'components';
+import {Icon} from '@thenewboston/ui';
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/containers/AppDetails/AppDetailsTopSection/Styles.ts:3]: `@thenewboston` import should be before local imports. Also, add a line break.

File: src/containers/AppStore/containers/AppDetails/AppDetailsTopSection/index.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+type Props = {
+  className?: string;
+  description: string;
+  logoUrl: string;
+  title: string;
+  websiteUrl: string;
+};
+
+const AppDetailsTopSection: FC<Props> = ({className, description, logoUrl, title, websiteUrl}) => {
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/containers/AppDetails/AppDetailsTopSection/index.tsx:16]: `FC` -> `SFC`

File: src/containers/AppStore/containers/AppDetails/AppDetailsTopSection/index.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+  const history = useHistory();
+  const [isWebsiteHovered, setIsWebsiteHovered] = useState(false);
+  const handleVisitWebsite = (): void => {
+    window.open(websiteUrl, '_blank', 'noopener noreferrer');
+  };
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/containers/AppDetails/AppDetailsTopSection/index.tsx:21]: add line breaks b/w hooks(3rd party and react's own) and functions; increases readability.

File: src/containers/AppStore/containers/AppStoreHome/AppStoreHero/Styles.ts [hunk 10]
```diff
@@ -0,0 +1,124 @@
...
+`;
+
+export const ImageOuter = styled.img`
+  max-height: 100%;
+  position: absolute;
+  right: 0;
+`;
+
+export const OrangeCircle = styled.div`
+  background-color: #febbbaf0;
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/containers/AppStoreHome/AppStoreHero/Styles.ts:85]: 8 chars in a color hex code? 😕 What does the last 2 mean?

File: src/containers/AppStore/containers/AppDetails/AppDetailsSlider/Styles.ts [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import styled from 'styled-components';
+import {Container as SharedContainer} from 'components';
+
+export const Container = styled.div`
+  background: rgba(246, 249, 252, 0.5);
```
Comment (thesanjeevsharma) [inline src/containers/AppStore/containers/AppDetails/AppDetailsSlider/Styles.ts:5]: The carousel background is white in Figma

File: src/containers/AppStore/containers/AppStoreHome/AppStoreHero/Styles.ts (hunk 10 above)
Comment (jamessspanggg) [reply src/containers/AppStore/containers/AppStoreHome/AppStoreHero/Styles.ts:85]: its for the alpha channel, i.e. opacity: https://stackoverflow.com/questions/6909896/hex-colors-in-android-are-some-times-8-digits-how-what-is-the-difference-betwe

I copied directly from the figma though

File: src/containers/AppStore/containers/AppDetails/AppDetailsSlider/Styles.ts (hunk 11 above)
Comment (jamessspanggg) [reply src/containers/AppStore/containers/AppDetails/AppDetailsSlider/Styles.ts:5]: ah yes, dk why i set that color, maybe designer changed it

File: src/containers/AppStore/components/AppCard/index.tsx (hunk 1 above)
Comment (jamessspanggg) [reply src/containers/AppStore/components/AppCard/index.tsx]: ah my bad, i created the `scss` version and migrated to styled-components afterwards, that's why there's some mistakes here and there, thanks for the comments

**Comment by thesanjeevsharma** [comment]
font-size for this is 12px in figma.

image

**Comment by thesanjeevsharma** [comment]
the heading and the cards should be aligned.

image

**Comment by thesanjeevsharma** [comment]
we have a separator here in the designs

image

**Comment by thesanjeevsharma** [comment]
I think we should get rid of the scroll bar here.

Also, the current image should not be faded and it has a border in design. Is that possible?

image

**Comment by thesanjeevsharma** [comment]
I think because of the `pre` tag overview did not take 100% width. Is this a fix for the front-end or did I make a mistake while entering the data?

image

**Comment by jamessspanggg** [comment]
> we have a separator here in the designs
> 
> image

Screen Shot 2021-09-11 at 9 29 01 AM

is it not showing on ur side?

**Comment by jamessspanggg** [comment]
@thesanjeevsharma fixed all ur comments except for the separator as it is visible?

**Comment by thesanjeevsharma** [comment]
@jamessspanggg take a look at the separator! Weird. 😄 

It appears and disappears as I resize the window. 

https://user-images.githubusercontent.com/29539278/132940834-979546d2-7d86-409a-8f3b-ef8cdebe88b6.mp4

**Review by thesanjeevsharma — APPROVED** [review]

**Comment by thesanjeevsharma** [comment]
🥇 Nice work!

**Pull request merged**